spatialmath/box.go

```diff
@@ -352,3 +353,100 @@ func separatingAxisTest(positionDelta, plane r3.Vector, halfSizeA, halfSizeB [3]
+	for i := 0.0; i <= b.halfSize[0]; i += iter {
+		for k := 0.0; k <= b.halfSize[2]; k += iter {
+			// front and back faces
+			p1 := []float64{i, b.halfSize[1], k}
+			p2 := []float64{i, b.halfSize[1], -k}
+			p3 := []float64{-i, b.halfSize[1], k}
+			p4 := []float64{-i, b.halfSize[1], -k}
+			p5 := []float64{-i, -b.halfSize[1], -k}
+			p6 := []float64{-i, -b.halfSize[1], k}
+			p7 := []float64{i, -b.halfSize[1], -k}
+			p8 := []float64{i, -b.halfSize[1], k}
+			if i == 0.0 && k == 0.0 {
+				faces = append(faces, p1, p5)
+			} else if i == 0.0 && k > 0.0 {
+				faces = append(faces, p1, p2, p7, p8)
+			} else if i > 0.0 && k == 0.0 {
+				faces = append(faces, p1, p3, p6, p7)
+			} else {
+				faces = append(faces, p1, p2, p3, p4, p5, p6, p7, p8)
+			}
+		}
+	}
+	for j := 0.0; j < b.halfSize[1]; j += iter {
+		for k := 0.0; k <= b.halfSize[2]; k += iter {
+			// left and right faces
+			p1 := []float64{b.halfSize[0], j, k}
+			p2 := []float64{b.halfSize[0], j, -k}
+			p3 := []float64{-b.halfSize[0], j, k}
+			p4 := []float64{-b.halfSize[0], j, -k}
+			p5 := []float64{-b.halfSize[0], -j, -k}
+			p6 := []float64{-b.halfSize[0], -j, k}
+			p7 := []float64{b.halfSize[0], -j, -k}
+			p8 := []float64{b.halfSize[0], -j, k}
+			if j == 0.0 && k == 0.0 {
+				faces = append(faces, p1, p5)
+			} else if j == 0.0 && k > 0.0 {
+				faces = append(faces, p1, p2, p3, p4)
+			} else if j > 0.0 && k == 0.0 {
+				faces = append(faces, p1, p3, p6, p7)
+			} else {
+				faces = append(faces, p1, p2, p3, p4, p5, p6, p7, p8)
+			}
+		}
+	}
+	for i := 0.0; i < b.halfSize[0]; i += iter {
+		for j := 0.0; j < b.halfSize[1]; j += iter {
+			// top and bottom faces
+			p1 := []float64{i, j, b.halfSize[2]}
+			p2 := []float64{i, j, -b.halfSize[2]}
+			p3 := []float64{-i, j, b.halfSize[2]}
+			p4 := []float64{-i, j, -b.halfSize[2]}
+			p5 := []float64{-i, -j, -b.halfSize[2]}
+			p6 := []float64{-i, -j, b.halfSize[2]}
+			p7 := []float64{i, -j, -b.halfSize[2]}
+			p8 := []float64{i, -j, b.halfSize[2]}
+			if i == 0.0 && j == 0.0 {
+				faces = append(faces, p1, p5)
+			} else if i == 0.0 && j > 0.0 {
+				faces = append(faces, p1, p2, p7, p8)
+			} else if i > 0.0 && j == 0.0 {
+				faces = append(faces, p1, p3, p6, p7)
+			} else {
+				faces = append(faces, p1, p2, p3, p4, p5, p6, p7, p8)
+			}
+		}
+	}
+	// translate points by offset and rotate
+	myList := transformPointsToPose(faces, b.Pose().Orientation().RotationMatrix().mat, b.pose.Point())
+	return myList
+}
+
+// todo: add function description
+func transformPointsToPose(points [][]float64, rotationMatrix [9]float64, pose r3.Vector) []r3.Vector {
+	myMat := mat.NewDense(3, 3, rotationMatrix[:])
+	var myList []r3.Vector
+	for i := 0; i < len(points); i++ {
+		pointMatrix := mat.NewVecDense(3, points[i])
+		actual := make([]float64, 3)
+		temp := mat.NewVecDense(3, actual)
+		temp.MulVec(myMat, pointMatrix)
+		myVec := r3.Vector{actual[0] + pose.X, actual[1] + pose.Y, actual[2] + pose.Z}
```

Comment:
This line, and everything above it in this `for` loop, can be replaced with the single line:

`myVec := Compose(NewPoseFromPoint(points[i]), Compose(NewPoseFromOrientation(r3.Vector{0,0,0}, pose.Orientation()), NewPoseFromPoint(pose.Point())))`

assuming the above change is made where `pose` is the full `Pose` object of the geometry